scanpointgenerator/core/dimension.py

```diff
@@ -15,22 +17,41 @@ class Dimension(object):
     Represents a single dimension within a scan.
     """
 
-    def __init__(self, generator):
-        self.axes = list(generator.axes)
-        """list(int): Unrolled axes within the dimension"""
+    def __init__(self, generators, excluders=None):
+        self.axes = list(axis for g in generators for axis in g.axes)
+        """list(str): Unrolled axes within the dimension"""
         self.size = None
         """int: Size of the dimension"""
-        self.upper = [generator.positions[a].max((0,)) for a in generator.axes]
+        self.upper = [g.positions[a].max((0,)) for g in generators for a in g.axes]
         """list(float): Upper bound for the dimension"""
-        self.lower = [generator.positions[a].min((0,)) for a in generator.axes]
+        self.lower = [g.positions[a].min((0,)) for g in generators for a in g.axes]
         """list(float): Lower bound for the dimension"""
-        self.alternate = generator.alternate
-        self.generators = [generator]
-        self._masks = []
-        self._max_length = generator.size
+        self.generators = list(generators)
+        self.excluders = list(excluders) if excluders is not None else []
+        self.alternate = generators[0].alternate
         self._prepared = False
         self.indices = []
 
+        # validate alternating constraints
+        started_alternating = False
+        for g in self.generators:
+            if started_alternating and not g.alternate:
+                raise ValueError(
+                        "Cannot nest non-alternating generators in "
+                        "alternating generators within a Dimension")
+            started_alternating = started_alternating or g.alternate
```

Comment:
Do we want to permit this use case? It may produce "unexpected" paths (a non-alternating generator will be run in its reverse direction every other run of the dimension).

Reply:
I think we should not allow this use case, but put your comment in the raised ValueError so people know why...